WIP: Look for L<> entries without sections

richsalz opened the pull request:
This is a WIP because

- I don't think the regexp's in find-doc-nits are optimal/right
- They are broken around the ```L<fooB<type>asdf>``` kind of thing.

Ping Richard Levitte, PhD (doctorate of perl hacking)

levitte commented:
The `$markup_re` I made is made exactly to match markup with inside markup. The `(?-1)` in that regexp is a regexp recursion point. If you go to the perlre manual and look up `(?R)`, you'll find the documentation on this.

... 
Darn, now I'm tempted to experiment. That'll be for Monday...

richsalz commented:
> Darn, now I'm tempted to experiment.

Nerd-snipe; https://xkcd.com/356/

richsalz commented:
Merged #10099 into this.  All "bad links" PR's are now in the bad-links* branches.

levitte commented on util/find-doc-nits:
```diff
@@ -444,6 +444,21 @@ sub check {
         check_section_location($id, $contents, "EXAMPLES", "SEE ALSO");
     }
 
+    # Check for proper in-man-3 API links.
```
This comment should probably be moved down to the next while.

richsalz replied:
fixed.  added a comment before that while, tho.

levitte commented on util/find-doc-nits:
```diff
@@ -444,6 +444,21 @@ sub check {
+        my $target = $1;
+        err($id, "Bad L<$target>")
+            unless $target =~ /^[_[:alpha:]][_[:alnum:]]*$/
+    }
```
More generally speaking, doesn't this mean that we simply do not allow links that doesn't match any of these regexps?

`m,L<https?://,`
`m,L</,`
`m,L<(?:.*?|)(.*?)\((\d)\)(?:/.*?)?>`

with the added caveat that, for the last one, we need to check if `$2 == 3`, and if it is, that `$1` has valid symbol syntax.

richsalz replied:
I don't know.  I do know that what is there works now.

richsalz commented:
Closing due to lack of interest; branch removal next week in case anyone wants to pick up some of the less-controversial changes.

levitte commented on util/find-doc-nits:
```diff
@@ -444,6 +444,22 @@ sub check {
+        my $target = $1;
+        next if $target =~ /^https?:/;
+        next if $target =~ m@^/@;
+        next if $target =~ m@\([1357]\)@;
```
The regexp should be `m@\([1357]\)$@` (ending $)

richsalz replied:
fixed.

levitte commented on util/find-doc-nits:
```diff
@@ -444,6 +444,22 @@ sub check {
+        next if $target =~ /^https?:/;
+        next if $target =~ m@^/@;
+        next if $target =~ m@\([1357]\)@;
+        next if $target =~ m@\([1357]\)/[^>]*>@;
```
`m@\([1357]\)/.*$@`

levitte commented:
A simple "ping" would have been enough. I leave it to you to revive this PR

richsalz commented:
As I said elsewhere, I thought there was no interest in the bad-links stuff overall and I was wrong.
Fixed the script, corrected some errors, rebased, squashed, pushed, re-opened.

levitte commented:
Mind you, this is still closed

richsalz commented:
I could not re-open this, so I made a new PR, #10240 
Every time I hear the phrase "mind you," I think of the Monty Python and the Holy Grail subtitles.

levitte commented:
Weird that you couldn't re-open this...
